draft-ietf-tls-tls13.md

```diff
@@ -2319,8 +2332,8 @@ Servers MUST NOT send a KeyShareEntry for any group not
 indicated in the "supported_groups" extension and
 MUST NOT send a KeyShareEntry when using the "psk_ke" PskKeyExchangeMode.
 If a HelloRetryRequest was received by the client, the client MUST verify that the
-selected NamedGroup matches one that the client sent in its supported_groups
-extension and MUST abort the handshake with an "illegal_parameter" alert otherwise.
+selected NamedGroup in the ServerHello is the same as that in the HelloRetryRequest. If this check
+fails, the client MUST abort the handshake with an "illegal_parameter" alert.
```

Comment:
I may be confused, but the graf directly above actually refers to the processing rules for the HRR itself and requires that it be consistent with the CH.

"Upon receipt of this extension in a HelloRetryRequest, the client MUST verify that (1) the selected_group field corresponds to a group which was provided in the "supported_groups" extension in the original ClientHello; and (2) the selected_group field does not correspond to a group which was provided in the "key_share" extension in the original ClientHello. If either of these checks fails, then the client MUST abort the handshake with an "illegal_parameter" alert. Otherwise, when sending the new ClientHello, the client MUST replace the original "key_share" extension with one containing only a new KeyShareEntry for the group indicated in the selected_group field of the triggering HelloRetryRequest."


This paragraph refers to the processing rules for the SH, so that's where we enforce consistency.